android/src/main/java/org/mozilla/mozstumbler/client/mapview/maplocation/LocationListenerWhileScanningOff.java

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+package org.mozilla.mozstumbler.client.mapview.maplocation;
+
+import android.location.LocationManager;
+
+import org.mozilla.mozstumbler.client.mapview.MapFragment;
+import org.mozilla.mozstumbler.client.mapview.maplocation.MapLocationListener.MapPositionUpdater;
+import org.mozilla.mozstumbler.client.mapview.maplocation.MapLocationListener.MapPositionUpdater.ReceivedLocationCallback;
+
+import java.lang.ref.WeakReference;
+
+public class LocationListenerWhileScanningOff {
+    private final LocationManager mLocationManager;
+    private final ReceivedLocationCallback mReceivedGpsLocation = new ReceivedLocationCallback() {
+        @Override
+        public void receivedLocation() {
+            // Once the gps begins receiving locations, stop the network listener
+            MapLocationListener.enableLocationListener(mLocationManager, false, mNetworkLocationListener);
+        }
+    };
+
+    private final MapPositionUpdater mNetworkLocationListener;
+    private final MapPositionUpdater mGpsLocationListener;
+
+    private final WeakReference<MapFragment> mMapFragment;
+
+    LocationListenerWhileScanningOff(LocationManager manager, MapFragment mapFragment) {
+        mLocationManager = manager;
+        mMapFragment = new WeakReference<MapFragment>(mapFragment);
+        mNetworkLocationListener = new MapPositionUpdater(mapFragment, LocationManager.NETWORK_PROVIDER, 1000, null);
+        mGpsLocationListener = new MapPositionUpdater(mapFragment, LocationManager.GPS_PROVIDER, 1000, mReceivedGpsLocation);
+        // ^ 1Hz is quite frequent, but it is only active when the map is being viewed (and scan off), so it is unlikely to pose a power problem
+    }
+
+    public void start() {
+        setActiveLocationUpdatesEnabled(true);
+    }
+
+    public void stop() {
+        setActiveLocationUpdatesEnabled(false);
+    }
+
+    private  void setActiveLocationUpdatesEnabled(boolean enable) {
+        if (mLocationManager == null) {
```

Comment:
Should we move - or at least have a check for the location manager being null in `LocationListenerWhileScanningOff` constructor?